pkg/identityserver/storetest/end_device_store.go

```diff
@@ -460,3 +462,81 @@ func (st *StoreTest) TestEndDeviceStorePagination(t *T) {
 		}
 	})
 }
+
+func (st *StoreTest) TestEndDeviceBatchUpdate(t *T) {
+	usr1 := st.population.NewUser()
+	app1 := st.population.NewApplication(usr1.GetOrganizationOrUserIdentifiers())
+
+	var all []*ttnpb.EndDevice
+	for i := 0; i < 3; i++ {
+		all = append(all, st.population.NewEndDevice(app1.GetIds()))
+	}
+
+	dev1 := all[0]
+	dev2 := all[1]
+	dev3 := all[2]
+
+	s, ok := st.PrepareDB(t).(interface {
+		Store
+		is.EndDeviceStore
+	})
+	defer st.DestroyDB(t, false)
+	defer s.Close()
+	if !ok {
+		t.Fatal("Store does not implement EndDeviceStore")
+	}
+
+	t.Run("BatchUpdateEndDeviceLastSeen", func(t *T) {
+		a, ctx := test.New(t)
+
+		validDevTime := time.Now()
+		dev1.LastSeenAt = ttnpb.ProtoTimePtr(validDevTime)
+		dev2.LastSeenAt = ttnpb.ProtoTimePtr(validDevTime)
+		dev3.LastSeenAt = ttnpb.ProtoTimePtr(validDevTime.Add(10 * time.Second))
+
+		batch := []*ttnpb.BatchUpdateEndDeviceLastSeenRequest_EndDeviceLastSeenUpdate{
+			{Ids: dev1.Ids, LastSeenAt: dev1.LastSeenAt},
+			{Ids: dev2.Ids, LastSeenAt: dev2.LastSeenAt},
+			{Ids: dev3.Ids, LastSeenAt: dev3.LastSeenAt},
```

Comment:
Should we also check that it doesn't crash if the same device is updated multiple times?